### PR TITLE
Cleanup location bundle

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR cleanup the location-bundle.

### CODE REVIEW

File: src/Sulu/Bundle/LocationBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -22,44 +22,23 @@ class Configuration implements ConfigurationInterface
...
-                        ->end()
-                    ->end()
-                ->end()
                 ->arrayNode('geolocators')
                     ->addDefaultsIfNotSet()
                     ->children()
                         ->arrayNode('nominatim')
                             ->addDefaultsIfNotSet()
                             ->children()
+                                ->scalarNode('api_key')->isRequired()->end()
```
Comment (nnatter) [inline src/Sulu/Bundle/LocationBundle/DependencyInjection/Configuration.php]: do we need to set the `api_key` of `google` to required too?

File: src/Sulu/Bundle/LocationBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -22,44 +22,23 @@ class Configuration implements ConfigurationInterface
     public function getConfigTreeBuilder()
     {
         $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('sulu_location')
+        $treeBuilder->root('sulu_location')
             ->children()
                 ->arrayNode('enabled_providers')
```
Comment (nnatter) [inline src/Sulu/Bundle/LocationBundle/DependencyInjection/Configuration.php]: i think we can remove this

File: src/Sulu/Bundle/LocationBundle/Resources/docs/index.md [hunk 3]
```diff
@@ -72,10 +40,11 @@ You can choose between at least two different endpoints:
 - OSM: http://nominatim.openstreetmap.org/search: [Example](http://nominatim.openstreetmap.org/search?format=json&q=466+dorchester+road+weymouth)
 - MapQuest: http://open.mapquestapi.com/nominatim/v1/search.php: [Example](http://open.mapquestapi.com/nominatim/v1/search.php?format=json&q=466+dorchester+road+weymouth)
 
-Both of the aboce services are free-to-use, but the mapquest endpoint seems faster.
+Both of the above services are free-to-use, but the mapquest endpoint seems faster.
 
 Configuration:
 
+- **api_key**: Optional, but recommended if making many requests ([Developer Mapquest](https://developer.mapquest.com/user/me/apps)).
 - **endpoint**: The endpoint to use
```
Comment (nnatter) [inline src/Sulu/Bundle/LocationBundle/Resources/docs/index.md]: i guess we can remove this?

File: src/Sulu/Bundle/LocationBundle/Resources/docs/index.md [hunk 4]
```diff
@@ -84,19 +53,4 @@ Configuration:
...
-$ grunt watch
-````
-
-### Developing the Location content type
-
-The location content type can be developed in isolation at the following URL
-(after a grunt build)
-
-http://sulu.lo/bundles/sululocation/js/examples/index.html
+- **api_key**: Optional, but recommended if making many requests ([Google Console](http://g.co/dev/maps-no-account)).
```
Comment (nnatter) [inline src/Sulu/Bundle/LocationBundle/Resources/docs/index.md]: i think this is not optional anymore?

File: src/Sulu/Bundle/LocationBundle/Resources/docs/index.md (hunk 3 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/LocationBundle/Resources/docs/index.md]: you could use another provider (this is the official)

File: src/Sulu/Bundle/LocationBundle/Resources/docs/index.md (hunk 4 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/LocationBundle/Resources/docs/index.md]: no :)

File: src/Sulu/Bundle/LocationBundle/Geolocator/GeolocatorLocation.php [hunk 5]
```diff
@@ -14,165 +14,153 @@
...
 
     /**
-     * @var string
+     * @var string|null
      */
     protected $country;
 
     /**
-     * @var float
+     * @var float|null
```
Comment (nnatter) [inline src/Sulu/Bundle/LocationBundle/Geolocator/GeolocatorLocation.php]: shouldnt locations at leas have a `longitude` and `latitude`?  🤔

File: src/Sulu/Bundle/LocationBundle/Resources/docs/index.md [hunk 6]
```diff
@@ -10,50 +10,18 @@ sulu_location:
     types:
```
Comment (nnatter) [inline src/Sulu/Bundle/LocationBundle/Resources/docs/index.md]: i think this is outdated?

File: src/Sulu/Bundle/LocationBundle/Geolocator/GeolocatorLocation.php (hunk 5 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/LocationBundle/Geolocator/GeolocatorLocation.php]: no for example polygon results have no long/lat

**Comment by alexander-schranz** [comment]
@wachterjohannes UPGRADE.md missing

File: src/Sulu/Bundle/LocationBundle/Geolocator/GeolocatorLocation.php [hunk 7]
```diff
@@ -14,165 +14,153 @@
...
     }
 
-    public function getCode()
+    public function getCode(): ?string
     {
         return $this->code;
     }
 
-    public function setCode($code)
+    public function setCode(?string $code): void
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/LocationBundle/Geolocator/GeolocatorLocation.php]: would add fluent setters like we always use

**Pull request merged**